### ISSUE TITLE
Darrowshire [AQ20] Ossirian crystals not working

### ISSUE THREAD

**Issue by Bloodmoon-Darrowshire** [LightsHope/server#1061]
Ossirian still retains his 300% buff when a crystal is activated.
Even when Ossirian is 2 feet away from the crystal it still does nothing.

**Comment by Road-block** [comment]
Old pull request / issue from Elysium that didn't carry over.  
I think it's relevant https://github.com/elysium-project/server/pull/615

**Comment by Road-block** [comment]
To provide some context this refers to this line https://github.com/LightsHope/server/blob/dcecf9555d3526d73ded9374ac90b826b74cea6f/src/scripts/kalimdor/silithus/ruins_of_ahnqiraj/boss_ossirian.cpp#L435 failing to return an instance of `ossirian` which in turn makes it so https://github.com/LightsHope/server/blob/dcecf9555d3526d73ded9374ac90b826b74cea6f/src/scripts/kalimdor/silithus/ruins_of_ahnqiraj/boss_ossirian.cpp#L465 fails to cast weakness on him and the problem cascades from there.  
https://github.com/LightsHope/server/blob/dcecf9555d3526d73ded9374ac90b826b74cea6f/src/scripts/kalimdor/silithus/ruins_of_ahnqiraj/boss_ossirian.cpp#L160 doesn't trigger and he stays in Supreme mode permanently.

**Comment by Road-block** [comment]
The workaround suggested in the pull request appears to be similar to what is already used for similar gameobjects out in the world.  
Example https://github.com/LightsHope/server/blob/dcecf9555d3526d73ded9374ac90b826b74cea6f/src/scripts/eastern_kingdoms/blasted_lands/blasted_lands.cpp#L299

**Comment by Road-block** [comment]
Just a note that his normal hit seems severely under-tuned (possibly to counteract the perma-supreme mode?)  
Numerous references from 2006 point at him hitting for 10-13k before armor mitigation in supreme mode and 19k for crushes (which maths out perfectly for the upper damage range)  
This would bring him down to ~2k hits when debuffed by crystals, on tanks (10k/3 * 0.6) .  

If he is left as is, then even fixing the actual mechanics will not stop most even moderately geared raids from simply killing him in supreme mode like they do now.  

I also remember his trash engaging when boss is pulled, but looking for actual video evidence of that.  
**As it stands now, with raids skipping all of his trash and him hitting for quite low damage while in Supreme mode**, he's the easiest boss in the instance.

**Comment by Road-block** [comment]
Example log showing an average of under 2k auto-attacks from Ossirian.  
[legacy-logs](https://legacy-logs.com/Vanilla/Raids/Evaluation/DmgTaken/?rid=4671&attempts=469691&player=44367&sel=0&pet=0&tarid=&mode=2)

Textual references from 2005/6.  
- http://web.archive.org/web/20061109203510/http://www.wow-tactics.de/20_taktiken/aq20/ossirian.php
- http://www.tentonhammer.com/guides/tentonhammers-guide-to-the-ruins-of-ahn-qiraj-section-2-6
- https://www.tacticalgamer.com/forum/mass-multiplayer/world-of-warcraft/world-of-warcraft-tactics-strategies-resources/23099-aq-20-boss-tactics
- https://www.tacticalgamer.com/forum/mass-multiplayer/world-of-warcraft/world-of-warcraft-tactics-strategies-resources/23099-aq-20-boss-tactics
- http://olivier.rs-team.org/strats_wow/AQ20OSSIRIAN.pdf
- http://leftoversraiding.org/index.php/topic/817/1/pulling-ossirian.html#post6739

Video from 2005/6 showing damage when weakened.
- [Tank PoV](https://youtu.be/Je2USKK2qhs?t=55)
- https://www.youtube.com/watch?v=Mlh8lSaBqPk
- https://www.youtube.com/watch?v=qb0zZ3xkKLk
- https://www.youtube.com/watch?v=Y2ITGk5_8hc

Basically: 
- his Warstomp ability seems to be ok (~3k on legacy-logs for supreme mode vs ~1k in textual and video evidence from 2006 for weakened state)
- his auto-attack seems to be heavily undertuned even if you account for the armor mitigation differences between an appropriately geared tank for the instance as shown in the classic videos and a full T2 one as shown in recent logs.  

His auto-attack should have a variance of 10.000-14.000 before armor mitigation and defensive stance modifier in supreme mode.

**Comment by Cabro** [comment]
I would also love to see people doing this boss as they should, not abusing with the single tank with trinket crap.
